### PR TITLE
Run back-reference breeze command as part of our CI builds

### PR DESCRIPTION
**Pull request by potiuk**
The `back-reference` command post-processes our docuementation to add redirections for new pages in the old documentation, so that user browsing the page can change version without hittin 404 error.

This command was not run in CI and some refactorings could break it (as it already happened in #33626 - fixed by #33790).

Most other breeze commands are already run in CI, so this one should be no different.

This PR adds back-reference generation as part of our regular documentation building step.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -645,6 +645,14 @@ jobs:
...
           --override-versioned
           ${{ needs.build-info.outputs.docs-filter-list-as-string }}
+      - name: "Generate back-reference for providers"
+        run: breeze release-management add-back-references all-providers
+      - name: "Generate back-reference for apache-airflow"
+        run: breeze release-management add-back-references apache-airflow
+      - name: "Generate back-reference for docker-stack"
+        run: breeze release-management add-back-references docker-stack
+      - name: "Generate back-reference for helm-chart"
+        run: breeze release-management add-back-references helm-chart
```
Comment (amoghrajesh) [inline .github/workflows/ci.yml]: small nit: for each command's title, can we do: `Generate back references for providers`, `Generate back references for apache-airflow`, and so on..

**Review by amoghrajesh — APPROVED** [review]
Makes sense to add this in the CI. One super nit, which you can feel free to ignore. Looks good +1

File: .github/workflows/ci.yml (hunk 1 above)
Comment (potiuk) [reply .github/workflows/ci.yml]: Sure. Let me just see if it succeeds with the current build and I will push a fixup

**Review by eladkal — APPROVED** [review]

**Comment by potiuk** [comment]
Looks good. Added @amoghrajesh suggestion. I also changed print() commands to use `get_console()` output that should format better the output on CI (the terminal provided by CI by default is not detected as supporting colors and log lines by rich so we have to use our console to force it).

**Pull request merged**